### PR TITLE
docs: add docs for secrets & jobs

### PR DESCRIPTION
**Pull request by npitsillos**
**Goals:**
- Adds documentation for the `jcloud` for the `secrets` and `jobs` support

- [x] check and update documentation. See [guide](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#-contributing-documentation) and ask the team.

### CODE REVIEW

File: docs/concepts/jcloud/index.md [hunk 1]
```diff
@@ -164,12 +196,14 @@ To access the [Grafana](https://grafana.com/)-powered dashboard, first get {ref}
...
+### List Flows, Secrets or Jobs
+
+#### List Flows
 
-To list all of your "Serving" Flows:
+To list all of your "Starting", "Serving", "Failed", "Updating", and "Paused" Flows:
 
 ```bash
-jc list
+jc list flows
```
Comment (deepankarm) [inline docs/concepts/jcloud/index.md:206]: Is this a breaking change? Does `jc list/update/remove` etc need params `flow/job/secret` etc now?

File: docs/concepts/jcloud/index.md [hunk 2]
```diff
@@ -82,6 +84,36 @@ jina flow --uses flow.yml
...
+#### Create a Secret
+
+To create a Secret for a Flow:
+
+```bash
+jc create secret mysecret -f rich-husky-af14064067 --from-literal "{'env-name': 'secret-value'}"
+```
+
+```{tip}
+You can optionally pass the `--update` flag to automatically update the Flow spec with the updated secret information. This flag will update the Flow which is hosted on the cloud. Finally, you can also optionally pass a Flow's yaml file path with `--path` to update the yaml file locally.
```
Comment (JoanFM) [inline docs/concepts/jcloud/index.md:98]: Link to the `env-from-secrets` parameter in Jina Docs. https://docs.jina.ai/cloud-nativeness/kubernetes/#deploy-flow-with-custom-environment-variables-and-secrets

File: docs/concepts/jcloud/index.md [hunk 3]
```diff
@@ -82,6 +84,36 @@ jina flow --uses flow.yml
...
+#### Create a Job
+
+To create a Job for a Flow:
+
+```bash
+jc create job job-name rich-husky-af14064067 image 'job entrypoint' --timeout 600 --backofflimit 2
+```
+
+```{tip}
+`image` can be any Executor image passed to a Flow's Executor `uses` or any normal docker image prefixed with `docker://`
```
Comment (JoanFM) [inline docs/concepts/jcloud/index.md:114]: in this case it is not true, unfortunately, an Executor image cannot be converted to a Flow. We have a ticket about this, but no time for it. https://github.com/jina-ai/jina/issues/5690

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: docs/concepts/jcloud/index.md (hunk 1 above)
Comment (npitsillos) [reply docs/concepts/jcloud/index.md:206]: Yes we need this to distinguish between the resource to create.

File: docs/concepts/jcloud/index.md (hunk 3 above)
Comment (npitsillos) [reply docs/concepts/jcloud/index.md:114]: You mean it cannot be converted to a job? I added tests for the images here https://github.com/jina-ai/jina-operator/blob/main/api/test/e2e/jobs_test.go and it is running fine adding the executor image.

Comment (JoanFM) [reply docs/concepts/jcloud/index.md:114]: but what is the entrypoint?

Comment (npitsillos) [reply docs/concepts/jcloud/index.md:114]: Keep in mind I am only running `jina` as the entrypoint and checking that the logs are not empty in this case.

**Review by JoanFM — APPROVED** [review]

File: docs/concepts/jcloud/index.md (hunk 1 above)
Comment (JoanFM) [reply docs/concepts/jcloud/index.md:206]: but can we have `flow` as default? to avoid breaking changes?

**Comment by npitsillos** [comment]
Let's not merge this until I update `jcloud` to support the resource as a sub-command. Currently the user should run `jc list jobs` however we will change it to `jc jobs list` instead. Same for the other resources

**Review by deepankarm — CHANGES_REQUESTED** [review]
image

The sections are weirdly structured. Please change it to resource-wise - one section each for `Flows`, `Jobs` & `Secrets`

**Pull request merged**